File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/dao/ImportedFile.java
package uk.gov.justice.digital.hmpps.riskprofiler.dao;

import lombok.*;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.atomic.AtomicInteger;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class ImportedFile<RiskDataSet> {
    private String fileName;
    private LocalDateTime fileTimestamp;
    private Map<String, RiskDataSet> dataSet;

    @Builder.Default
    private AtomicInteger index = new AtomicInteger();
    @Builder.Default
    private AtomicInteger linesProcessed = new AtomicInteger();
    @Builder.Default
    private AtomicInteger linesDup = new AtomicInteger();
    @Builder.Default
    private AtomicInteger linesError = new AtomicInteger();
    @Builder.Default
    private AtomicInteger linesInvalid = new AtomicInteger();

    public void reset() {
        dataSet = new HashMap<>();
        index.set(0);
        linesProcessed.set(0);
        linesDup.set(0);
        linesError.set(0);
        linesInvalid.set(0);
    }

}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/dao/ImportedFile.kt
package uk.gov.justice.digital.hmpps.riskprofiler.dao

import uk.gov.justice.digital.hmpps.riskprofiler.datasourcemodel.FileType
import java.time.LocalDateTime
import java.util.concurrent.atomic.AtomicInteger

data class ImportedFile<RiskDataSet>(
  var fileName: String? = null,
  var fileTimestamp: LocalDateTime? = null,
  var dataSet: MutableMap<String, RiskDataSet>? = null,
  var fileType: FileType? = null,
  val index: AtomicInteger = AtomicInteger(),
  val linesProcessed: AtomicInteger = AtomicInteger(),
  val linesDup: AtomicInteger = AtomicInteger(),
  val linesError: AtomicInteger = AtomicInteger(),
  val linesInvalid: AtomicInteger = AtomicInteger(),
) {
  fun reset() {
    dataSet = HashMap()
    index.set(0)
    linesProcessed.set(0)
    linesDup.set(0)
    linesError.set(0)
    linesInvalid.set(0)
  }
}
